`setup.py` incorrectly disables CFFI extension build for PyPy

The `setup.py` script currently claims that no extension module should be built for PyPy:

https://github.com/jnwatson/py-lmdb/blob/4e5c2ebf47ff1da60c01254bfdcac116c893abfc/setup.py#L183-L185

This is incorrect. CFFI uses extension modules on PyPy just the same as on CPython. Installing `lmdb` without the module built just means that CFFI will attempt to build it on first import, and fail given that the sources are missing:

```console
$ uv venv -p pypy3.11
Using PyPy 3.11.13 interpreter at: /usr/bin/pypy3.11
Creating virtual environment at: .venv
Activate with: source .venv/bin/activate
$ uv pip install lmdb
Resolved 3 packages in 4ms
░░░░░░░░░░░░░░░░░░░░ [0/3] Installing wheels...
Installed 3 packages in 4ms
 + cffi==2.0.0
 + lmdb==1.7.5
 + pycparser==2.23
$ .venv/bin/python -c 'import lmdb'
.venv/lib/pypy3.11/site-packages/lmdb/__pycache__/lmdb_cffi.c:61:14: fatal error: lmdb.h: No such file or directory
   61 |     #include "lmdb.h"
      |              ^~~~~~~~
compilation terminated.
Traceback (most recent call last):
  File "/tmp/.venv/lib/pypy3.11/site-packages/lmdb/__init__.py", line 42, in <module>
    from lmdb.cpython import *
ModuleNotFoundError: No module named 'lmdb.cpython'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/pypy3.11/distutils/unixccompiler.py", line 117, in _compile
    self.spawn(compiler_so + cc_args + [src, '-o', obj] +
  File "/usr/lib/pypy3.11/distutils/ccompiler.py", line 910, in spawn
    spawn(cmd, dry_run=self.dry_run)
  File "/usr/lib/pypy3.11/distutils/spawn.py", line 91, in spawn
    raise DistutilsExecError(
distutils.errors.DistutilsExecError: command '/usr/bin/gcc' failed with exit code 1

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/pypy3.11/cffi/ffiplatform.py", line 48, in _build
    dist.run_command('build_ext')
  File "/usr/lib/pypy3.11/distutils/dist.py", line 985, in run_command
    cmd_obj.run()
  File "/usr/lib/pypy3.11/distutils/command/build_ext.py", line 344, in run
    self.build_extensions()
  File "/usr/lib/pypy3.11/distutils/command/build_ext.py", line 453, in build_extensions
    self._build_extensions_serial()
  File "/usr/lib/pypy3.11/distutils/command/build_ext.py", line 478, in _build_extensions_serial
    self.build_extension(ext)
  File "/usr/lib/pypy3.11/distutils/command/build_ext.py", line 533, in build_extension
    objects = self.compiler.compile(sources,

              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/pypy3.11/distutils/ccompiler.py", line 574, in compile
    self._compile(obj, src, ext, cc_args, extra_postargs, pp_opts)
  File "/usr/lib/pypy3.11/distutils/unixccompiler.py", line 120, in _compile
    raise CompileError(msg)
distutils.errors.CompileError: command '/usr/bin/gcc' failed with exit code 1

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/tmp/.venv/lib/pypy3.11/site-packages/lmdb/__init__.py", line 48, in <module>
    from lmdb.cffi import *
  File "/tmp/.venv/lib/pypy3.11/site-packages/lmdb/cffi.py", line 372, in <module>
    _lib = _ffi.verify(_CFFI_VERIFY,

           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/pypy3.11/cffi/api.py", line 468, in verify
    lib = self.verifier.load_library()
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/pypy3.11/cffi/verifier.py", line 105, in load_library
    self._compile_module()
  File "/usr/lib/pypy3.11/cffi/verifier.py", line 201, in _compile_module
    outputfilename = ffiplatform.compile(tmpdir, self.get_extension())
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/pypy3.11/cffi/ffiplatform.py", line 20, in compile
    outputfilename = _build(tmpdir, ext, compiler_verbose, debug)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/pypy3.11/cffi/ffiplatform.py", line 54, in _build
    raise VerificationError('%s: %s' % (e.__class__.__name__, e))
cffi.VerificationError: CompileError: command '/usr/bin/gcc' failed with exit code 1
```

--- a/setup.py
+++ b/setup.py
@@ -181,6 +181,12 @@ else:
     print('Using cffi extension.')
     install_requires = ['cffi>=0.8; implementation_name=="cpython"']
     print('Using cffi, building extension module.')
+    # Ensure the source directory is on sys.path so that `import lmdb.cffi`
+    # works in build-isolated environments where pip runs setup.py from a
+    # temporary directory.
+    _source_dir = os.path.dirname(os.path.abspath(__file__))
+    if _source_dir not in sys.path:
+        sys.path.insert(0, _source_dir)
     try:
         import lmdb.cffi
         ext_modules = [lmdb.cffi._ffi.verifier.get_extension()]

